src/main/java/Duke.java

```diff
@@ -1,10 +1,149 @@
+        String deadlineInput;
+        try {
+            deadlineInput = task.trim().split("deadline")[1];
+        } catch (IndexOutOfBoundsException e) {
+            printWithLines("☹ OOPS!!! The description of deadline cannot be empty.");
+            throw new DukeException();
+        }
+        taskList.add(new Deadline(task.substring("deadline".length(), task.indexOf("/by")), task.substring(task.indexOf("/by") + "/by".length())));
+    }
+
+    private static void addEvent(String task) throws DukeException {
+        String eventInput;
+        try {
+            eventInput = task.trim().split("event")[1];
+        } catch (IndexOutOfBoundsException e) {
+            printWithLines("☹ OOPS!!! The description of event cannot be empty.");
+            throw new DukeException();
+        }
+        taskList.add(new Event(task.substring("event".length(), task.indexOf("/at")), task.substring(task.indexOf("/at") + "/at".length())));
+    }
+
+    public static void listTasks() {
+        System.out.println("____________________________________________________________");
+        int i = 1;
+        for (Task item: taskList) {
+            System.out.println((i++) + ". " + item);
+        }
+        System.out.println("____________________________________________________________");
+    }
+
+    public static void markTaskAsDone(String task) {
+        int taskIndex = Integer.parseInt(task.replace("done ", "")) -  1;
+        if (taskIndex > taskCount - 1) {
+            printWithLines(" The task " + (taskIndex + 1) + " doesn't exist.\nMake sure a valid task number is entered.");
+            return;
+        }
+        Task current = taskList.get(taskIndex);
+        current.markAsDone();
+        printWithLines("Nice! I've marked this task as done:\n" + current.toString());
+    }
+
+    public static void printWithLines(String output) {
+        String horizontalLine = "____________________________________________________________";
+        System.out.println(horizontalLine);
+        System.out.println(output);
+        System.out.println(horizontalLine);
+    }
+
+    public static void main(String[] args) throws DukeException {
+        taskList = new ArrayList<Task>(100);;
+        taskCount = 0;
+        String userInput;
         String logo = " ____        _        \n"
                 + "|  _ \\ _   _| | _____ \n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
         System.out.println("Hello from\n" + logo);
+        printWithLines("Hello! I'm Duke\n" + "What can I do for you?");
+        try{
+            taskList = Storage.loadData();
+        } catch(FileNotFoundException f) {
+            System.out.println("Error: save file not found");
+        }
+        Scanner in = new Scanner(System.in);
+        userInput = in.nextLine();
+        while (!userInput.equals("bye"))
+        {
```

Comment:
The open `{` should be on the same line as the while statement, as we are following the K&R style for brackets